core/comments/comment_view.ts

```diff
@@ -47,103 +47,148 @@ export class CommentView implements IRenderedElement {
       'class': 'blocklyComment blocklyEditable',
     });
 
-    this.topBar = dom.createSvgElement(
+    ({
+      topBar: this.topBar,
+      deleteIcon: this.deleteIcon,
+      foldoutIcon: this.foldoutIcon,
+      textPreview: this.textPreview,
+      textPreviewNode: this.textPreviewNode,
+    } = this.createTopBar(this.svgRoot, workspace));
+
+    ({foreignObject: this.foreignObject, textArea: this.textArea} =
+      this.createTextArea(this.svgRoot));
+
+    this.resizeHandle = this.createResizeHandle(this.svgRoot, workspace);
+
+    // TODO: Remove this comment before merging.
+    // I think we want comments to exist on the same layer as blocks.
+    workspace.getLayerManager()?.append(this, layers.BLOCK);
```

Comment:
The disadvantage of being on the same layer as blocks is that a block can go over the comment and look like it's inside it. That's a bigger issue with mutators bubbles, since they always hold blocks. I'm neutral on which one we should do for comments.

Reply:
There is also a weird thing where when we deserialize, the comments will be deserialized after blocks, so they'll always end up on top after load even if they were behind blocks before.